### DISCUSSION TITLE
router path consider radix tree ?

### DISCUSSION THREAD

**Discussion by me-v2** [encode/starlette#1939] [Ideas]
The starting point for issues should usually be a discussion...

https://github.com/encode/starlette/discussions

Possible bugs may be raised as a "Potential Issue" discussion, feature requests may be raised as an "Ideas" discussion. We can then determine if the discussion needs to be escalated into an "Issue" or not.

This will help us ensure that the "Issues" list properly reflects ongoing or needed work on the project.

---

- [ ] Initially raised as discussion #...

**Comment by adriangb** [comment]
Are you asking why we don't use a radix tree as the router data structure?

**Reply by me-v2** [reply]
`https://github.com/gin-gonic/gin/blob/master/tree.go`
`https://en.wikipedia.org/wiki/Radix_tree`

**Reply by Kludex** [reply]
:point_up_2: ❓

**Comment by Kludex** [comment]
Closing this since OP is not clear.
